src/modules/staking/services/staking.compute.service.ts

```diff
@@ -211,4 +213,76 @@ export class StakingComputeService {
                   .dividedBy(constantsConfig.MAX_PERCENT)
                   .toFixed();
     }
+
+    async computeOptimalCompoundFrequency(
+        stakeAddress: string,
+        positionAmount: string,
+        timeHorizon: number,
+    ): Promise<OptimalCompoundModel> {
+        const [apr, farmingToken] = await Promise.all([
+            this.stakeFarmAPR(stakeAddress),
+            this.stakingService.getFarmingToken(stakeAddress),
+        ]);
+
+        const denominatedAmount = denominateAmount(
+            positionAmount,
+            farmingToken.decimals,
+        ).toNumber();
+
+        const farmingTokenPrice = await this.tokenGetter.getDerivedEGLD(
+            farmingToken.identifier,
+        );
+        const egldPriceFarmingToken = new BigNumber(1).dividedBy(
+            farmingTokenPrice,
+        );
+        const transactionFeesInFarmingToken = new BigNumber(
+            constantsConfig.COMPOUND_TRANSACTION_FEE,
+        )
+            .multipliedBy(egldPriceFarmingToken)
+            .toNumber();
+
+        // Express compound iterations in minutes
+        const compoundIterations =
+            constantsConfig.COMPOUND_INTERVAL_DAYS * 24 * 60;
+
+        let optimalCompoundIterations = 0;
+        let optimalProfit = 0;
+
+        for (let iterator = 1; iterator < compoundIterations; iterator += 1) {
+            const rewards =
+                (1 +
+                    (parseFloat(apr) * timeHorizon) /
```

Comment:
In numerator of `rewards`, `apr` is a yearly APR, so in the denominator we should divide by 365 days. Here we are fine because `COMPOUND_INTERVAL_DAYS` has value `365`, but it is not explicit and if this value is changed in the future, the computation would be wrong. Maybe we could use `365` explicitly here in the denominator? Or have it as a constant `ONE_YEAR_IN_YEARS = 365`?

Reply:
done